src/lib/components/Creatibutors/CreatibutorsModal.js

```diff
@@ -183,6 +187,124 @@ export class CreatibutorsModal extends Component {
+        return identifier.scheme === 'orcid';
+      });
+
+      let aff_names = '';
+      creatibutor.affiliations.forEach((affiliation, idx) => {
+        aff_names += affiliation.name;
+        if (idx < creatibutor.affiliations.length - 1) {
+          aff_names += ', ';
+        }
+      });
+
+      return {
+        text: creatibutor.name,
+        value: orcid.identifier,
+        extra: creatibutor,
+        key: creatibutor.id,
+        content: (
+          <Header>
+            <Header.Content>
+              {creatibutor.name} (
+              <Image
+                src="/static/images/orcid.svg"
+                className="small-icon"
+                verticalAlign="middle"
+              />
+              {orcid.identifier})
+              <a
+                href={`https://orcid.org/${orcid.identifier}`}
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                <Icon link name="external alternate" className="spaced-left" />
+              </a>
+            </Header.Content>
+            <Header.Subheader>{aff_names}</Header.Subheader>
+          </Header>
+        ),
+      };
+    });
+
+    results.push({
+      text: 'Manual entry',
+      value: 'Manual entry',
+      extra: 'Manual entry',
+      key: 'manual-entry',
+      content: (
+          <Header>
+            <Header.Content>
+              <p>Couldn't find your person? You can <a>create a new entry</a></p>
+            </Header.Content>
+          </Header>
+      ),
+    });
+    return results;
+  }
+
+  onPersonSearchChange = ({ event, data, formikProps }, selectedSuggestions) => {
+    if (selectedSuggestions[0].key === 'manual-entry') {
+      this.setState({
+        showPersonForm: true
+      })
+      return
+    }
+
+    this.setState({
+      showPersonForm: true
+    }, () => {
+      const identifiers =
+        selectedSuggestions[0].extra.identifiers.map(
+          (identifier) => {
+            return identifier.identifier;
+          }
+        );
+      const affiliations =
+        selectedSuggestions[0].extra.affiliations.map(
+          (affiliation) => {
+            return affiliation;
+          }
+        );
+
+      const personOrOrgPath = `person_or_org`;
+      const familyNameFieldPath = `${personOrOrgPath}.family_name`;
+      const givenNameFieldPath = `${personOrOrgPath}.given_name`;
+      const identifiersFieldPath = `${personOrOrgPath}.identifiers`;
+      const affiliationsFieldPath = 'affiliations';
+
+      let chosen = {
+        [givenNameFieldPath]: selectedSuggestions[0].extra.given_name,
+        [familyNameFieldPath]: selectedSuggestions[0].extra.family_name,
+        [identifiersFieldPath]: identifiers,
+        [affiliationsFieldPath]: affiliations,
+      };
+      Object.entries(chosen).forEach(([path, value]) => {
+        formikProps.form.setFieldValue(path, value);
+      });
+      // Update identifiers render
+      this.identifiersRef.current.setState({
```

Comment:
From a quick look (but I am probably wrong), in general this is a sign that the state should be handled by the upper component.
If you remove the state `selectedOptions` from `CreatibutorsIdentifiers`, and you manage it in this cmp, you should have here:
```
this.setState({selectedIdentifer: ....})
```
in the render method, you pass it as a prop, so that `CreatibutorsIdentifiers` will re-render with this new `selectedIdentifier`. You will also handle the callbacks and formik set in the upperCmp: the inner one will become very dummy.

If the logic in this upper becomes very messy, you can always introduce a middle cmp, between the 2, where the middle one encapsulates/manages the state of the inner one and the upper one is a bit lighter...
I had to do something similar in the PIDs cmp.

Happy to chat IRL :) I hope it helps (even if I am very positive that you have already thought about this).